src/components/utils/Markdown.tsx

```diff
@@ -0,0 +1,165 @@
+  }
+
+  h2 {
+    font-size: 1.5em;
+  }
+
+  h3 {
+    font-size: 1.17em;
+  }
+
+  h4 {
+    font-size: 1em;
+  }
+
+  h5 {
+    font-size: 0.83em;
+  }
+
+  h6 {
+    font-size: 0.67em;
+  }
+
+  hr {
+    background-color: #000000;
+    height: 2px;
+    border: 0px;
+  }
+`;
+
+export interface MarkdownProps {
+  text: string;
+}
+
+const TitleBar = ({ language, code }: { language?: string; code: string }) => {
+  return (
+    <Row
+      mainAxisAlignment="flex-start"
+      crossAxisAlignment="stretch"
+      expand
+      justifyContent="space-between"
+      alignItems="center"
+      px="10px"
+      py="5px"
+      backgroundColor="#f5f5f5"
+      borderBottom="1px solid #eee"
+      borderRadius="6px"
+    >
+      <Box>{language || "plaintext"}</Box>
+      <CopyCodeButton code={code} />
+    </Row>
+  );
+};
+
+const CopyCodeButton = ({ code }: { code: string }) => {
+  const [copied, setCopied] = useState(false);
+
+  const handleCopyButtonClick = async (e: MouseEvent) => {
+    e.stopPropagation(); // Prevent this from triggering edit mode in the parent.
+
+    if (await copySnippetToClipboard(code)) setCopied(true);
+  };
+
+  useEffect(() => {
+    if (copied) {
+      const timer = setTimeout(() => setCopied(false), 2000);
+      return () => clearTimeout(timer);
+    }
+  }, [copied]);
+
+  return (
+    <Button
+      // TODO fix this
+      // @ts-ignore
+      onClick={handleCopyButtonClick}
+      size="xs"
+      variant="ghost"
+      px="5px"
+      _hover={{ background: "#EEEEEE" }}
+    >
+      <CopyIcon boxSize={4} mr={1} /> {copied ? "Copied!" : "Copy Code"}
+    </Button>
+  );
+};
+
+export function Markdown({ text }: { text: string }) {
+  const markdown = useMemo(
+    () => (
+      <StyledMarkdownWrapper>
+        <ReactMarkdown
+          remarkPlugins={[remarkGfm]} // supports github flavored markdown.
+          components={{
+            code({ node, inline, className, children, ...props }) {
+              const match = /language-(\w+)/.exec(className || "");
+              const code = String(children);
+              return !inline ? (
+                <Box
+                  padding="0"
+                  borderRadius="0.25rem"
+                  overflow="hidden"
+                  css={{
+                    "> div": { margin: "0 !important" },
+                    ".fa": { fontStyle: "normal !important" },
+                  }}
+                  {...props}
+                >
+                  <TitleBar language={match?.[1]} code={code} />
+                  <SyntaxHighlighter
+                    children={code}
+                    style={coy as any}
+                    language={match?.[1] || "text"}
+                    PreTag="div"
+                    wrapLongLines
+                    {...props}
+                  />
+                </Box>
+              ) : (
+                <code className={className} {...props}>
+                  {children}
+                </code>
```

Comment:
can we give this a grey background so it looks like it does on gh?

<img width="145" alt="Screenshot 2023-04-03 at 1 27 22 PM" src="https://user-images.githubusercontent.com/26209401/229619905-a2d33d3a-b913-47ca-8bff-298f49ab587e.png">

Reply:
Updating it all to use the Chakra <Code /> block instead, which solves this even with plaintext.